qiskit/circuit/quantumcircuit.py

```diff
@@ -2854,7 +2854,14 @@ def _assign_parameter(self, parameter: Parameter, value: ParameterValueType) ->
                     new_param = assignee.assign(parameter, value)
                     # if fully bound, validate
                     if len(new_param.parameters) == 0:
-                        instr.params[param_index] = instr.validate_parameter(new_param)
```

Comment:
Would it be more suitable to move this logic into `Instruction.validate_parameter`, to make it more generally accessible?

Reply:
I think the current form is better - putting `Parameter`-specific handling and normalisation in `validate_parameter` would be overloading the method, which I think would be quite surprising.  Also, several instructions (esp user ones) override `Instruction.validate_parameter`, but we'd want this to happen anyway.